intera_interface/src/motion_interface/interaction_options.py

```diff
@@ -0,0 +1,244 @@
+    ensure_path_to_file_exists,
+    clamp_float_warn
+)
+from rospy_message_converter import message_converter
+import yaml
+
+class InteractionOptions(object):
+    """
+    This class is a wrapper for the intera message:  InteractionControlCommand.msg
+    It's primary purpose is to facilitate message creation and input checking.
+    """
+
+    n_dim_cart = 6 # number of dimensions in Cartesian space
+    n_dim_joint = 7
+
+    # constants for interaction control modes
+    force_mode = InteractionControlCommand.FORCE_MODE
+    impedance_mode = InteractionControlCommand.IMPEDANCE_MODE
+    impedance_limit_mode = InteractionControlCommand.IMPEDANCE_WITH_FORCE_LIMIT_MODE
+    force_limit_mode = InteractionControlCommand.FORCE_WITH_MOTION_LIMIT_MODE
+
+    def __init__(self, header = None,
+                 interaction_control_active = None,
+                 K_impedance = None,
+                 max_impedance = None,
+                 K_nullspace = None,
+                 force_command= None,
+                 interaction_frame = None,
+                 endpoint_name = None,
+                 in_endpoint_frame = None,
+                 interaction_control_mode = None):
+        """
+        Create a interaction options object. All parameters are
+        optional. If ommitted or set to None, then use default value.
+        """
+        self._data = InteractionControlCommand()
+        self.set_header(header)
+        self.set_interaction_control_active(interaction_control_active)
+        self.set_K_impedance(K_impedance)
+        self.set_max_impedance(max_impedance)
+        self.set_K_nullspace(K_nullspace)
+        self.set_force_command(force_command)
+        self.set_interaction_frame(interaction_frame)
+        self.set_endpoint_name(endpoint_name)
+        self.set_in_endpoint_frame(in_endpoint_frame)
+        self.set_interaction_control_mode(interaction_control_mode)
+
+    def set_header(self, header = None):
+        if header is None:
+            self._data.header.stamp = rospy.get_rostime()
+            self._data.header.seq = 1
+            self._data.header.frame_id = "base"
+        else:
+            self._data.header = header
+
+    def set_interaction_control_active(self, interaction_active = None):
+        """
+        @param interaction_control_active:
+        --> None:  set it to True by default
+        --> [bool]: copy the element.
+        """
+        if interaction_active is None:
+            self.set_interaction_control_active(True) # default value
+        else:
+            self._data.interaction_control_active = interaction_active
+
+    def set_K_impedance(self, K_impedance = None):
+        """
+        @param K_impedance (Cartesian stiffness):
+        --> None:  populate with vector of default values
+        --> [float]:  copy all elements. Checks length.
+        """
+        if K_impedance is None:
+            self._data.K_impedance = [1300.0, 1300.0, 1300.0, 30.0, 30.0, 30.0]
```

Comment:
Yes, the fields correspond to [x, y, z, roll, pitch, yaw]. They are better defined in the message definition and script help.